packages/verified-fetch/src/index.ts

```diff
@@ -840,6 +840,7 @@ export async function createVerifiedFetch (init?: Helia | CreateVerifiedFetchIni
       dns,
       routers
     })
+    init.logger.forComponent('helia:verified-fetch').trace('created verified-fetch with libp2p config:', JSON.stringify(libp2pConfig))
```

Comment:
In general you want to avoid doing work like this for logging - the `JSON.stringify` will get invoked regardless of if logging is enabled or not.  It's not a big deal here since it's a one-off operation but it can degrade performance if you do it in a hot codepath.

Instead you can use `enabled`:

```js
import { enabled } from '@libp2p/logger'

if (enabled('helia:verified-fetch:trace')) {
  log.trace('expensive operation result: %s', JSON.stringify(libp2pConfig))
}
```

You should also include the `%s` token, otherwise it can mess up the formatting - in this case the `ms` diff since the last log will get printed before the `JSON.stringify` output.

Reply:
You could also use the `%j` formatter here instead and skip the conditional:

```suggestion
    init.logger.forComponent('helia:verified-fetch').trace('created verified-fetch with libp2p config: %j', libp2pConfig)
```